src/main/java/com/google/sps/servlets/ExamServlet.java

```diff
@@ -0,0 +1,180 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.sps.data.ExamClass;
+import com.google.sps.data.QuestionClass;
+import java.io.IOException;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.gson.Gson;
+import java.util.List;
+import java.util.ArrayList;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * Servlet to display the requested exam questions and send responses to ExamResponseServlet.
+ *
+ * @author  Aidan Molloy
+ */
+@WebServlet("/exam")
+public class ExamServlet extends HttpServlet{
+  /**
+   * Gets the exam ID from the httpRequest
+   *
+   * Arguments: 
+   *  request: provides request information from the HTTP servlet.
+   *  response: response object where servlet will write information to.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException{
+    response.setContentType("text/html;");
+    String examID = getParameter(request, "examID", "1");
```

Comment:
Is "1" the default parameter here? Why do we need a default exam id?

Reply:
There is now no default and it checks if an id exists.